app/docker/containers/queries.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from 'react-query';
+
+import { EnvironmentId } from '@/portainer/environments/types';
+import { error as notifyError } from '@/portainer/services/notifications';
+
+import { queryContainers } from './containers.service';
+
+export function useContainers(environmentId: EnvironmentId, filters?: object) {
+  return useQuery(
+    ['environments', environmentId, 'docker', 'containers', 'filters', filters],
```

Comment:
we can also use objects

```suggestion
    ['environments', environmentId, 'docker', 'containers', {filters}],
```